### DISCUSSION TITLE
Exception: No network is available.

### DISCUSSION THREAD

**Discussion by KRookoo1** [jchristn/WatsonWebsocket#72] [Ideas]
Hello.
When in computer not have ane real active network adapter or a computer is offline (any network adapter not has connection to networt) WatsonWebsocket has problem.

In function Start() in calss  WatsonWsClient first lin is:
if (!NetworkInterface.GetIsNetworkAvailable()) throw new InvalidOperationException("No network is available.");

NetworkInterface.GetIsNetworkAvailable() indicates whether any network connection is available and return true if a network connection is available.
If I use WatsonWebClient to localhost connection ( example WatsonWsClient("localhost",9001, false)) I havn't need internet connection or any network available.
Function NetworkInterface.GetAllNetworkInterfaces() return one network "LocalLoop" and it is work.

Please analize this problem and If client try start with localhost maybe you not need check networkAvailable?

**Comment by jchristn** [comment]
Hi @KRookoo1 are you using the library on a machine with no network connection?

**Reply by KRookoo1** [reply]
Yes and no... 
I use library to communicate with many process on one machine. 
This machine sometimes has internet connection bat not always.
But for library internet connection is not nessesary becaus this now communicate only in localhost (use loopback network connection)... mayby in futer I extend my projekt to communicate to many machine... but now only in localhost.

**Reply by jchristn** [reply]
Did you test to see if that exception would actually be thrown?

If you're using localhost your machine at minimum has the loopback adapter with TCP/IP, and what I've read is it always returns true if there is a network interface (loopback adapter is considered as such).

**Reply by KRookoo1** [reply]
Yes I tested two situation on two difrent machines:

1. I disble all network interface:
![image](https://user-images.githubusercontent.com/90908357/134123039-1f2f5215-b96f-4f4e-9dd8-27c500304ba9.png)
Ipconfig result:
![image](https://user-images.githubusercontent.com/90908357/134123189-e0495467-be18-4830-b17e-85d1ffda0155.png)

2.Secend situation when I have one interface LAN but cable is plug-off:
![image](https://user-images.githubusercontent.com/90908357/134123448-85332278-4caf-4dbf-8170-8df976c0a296.png)
Ipconfig:
![image](https://user-images.githubusercontent.com/90908357/134123483-a4362e38-b6e7-4432-8710-5311997f5f75.png)

In two situation function Start throw exception:
![image](https://user-images.githubusercontent.com/90908357/134123630-e7500787-2962-40b4-b740-02d5f2da569f.png)


But in microsoft documentation about NetworkInterface.GetIsNetworkAvailable:
https://docs.microsoft.com/en-us/dotnet/api/system.net.networkinformation.networkinterface.getisnetworkavailable?view=net-5.0
is writed:
Indicates whether any network connection is available.
and remarks:
A network connection is considered to be available if any network interface is marked "up" and **is not a loopback** or tunnel interface.

**Reply by jchristn** [reply]
Got it.  Thanks for all of the info.  I'm publishing v2.3.1 right now with this check removed.

**Reply by jchristn** [reply]
Here they are!  Thanks again @KRookoo1 

NuGet: https://www.nuget.org/packages/WatsonWebsocket/2.3.1
Commit: https://github.com/jchristn/WatsonWebsocket/commit/ba51a28111f3343873e6471a4c7cc22cf19a3a79
